### DISCUSSION TITLE
Connecting to webpage

### DISCUSSION THREAD

**Discussion by pseddonxvi** [nebhead/garage-zero#30] [Q&A] [answered]
Hi there, I'm trying to set up Garage Zero on my Pi Zero W and I am having issues connecting to the webpage, I can ping the Pi and SSH into it but I get an "ERR_CONNECTION_REFUSED" when trying to access it through a web browser using 192.168.1.142:9001. I did the automatic software install and everything worked fine, any ideas as to what may be causing this?

**Comment by nebhead** [comment]
What version of the OS are you running on your Pi Zero W?  

There _might_ have been some errors during the install that zipped past without seeing them.  Since even the supervisor is not responding, that would indicate to me that something didn't quite go right.  Before we try to re-install, let me know what OS you are using and we can make sure it's expected to work.

**Reply by pseddonxvi** [reply]
I am running the Raspberry Pi OS (Legacy) Lite version from 12/05/23 
![Screenshot 2024-01-20 140728](https://github.com/nebhead/garage-zero/assets/146755532/be7b08bb-2ffe-4058-a365-c53a8f2e7393)

**Comment by nebhead** [comment]
Probably Raspberry Pi OS Bullseye then.  Frankly, it _should_ work just fine, but we can try to repair the install by running some of the commands again.  

```bash
sudo apt install python3-dev python3-pip python3-rpi.gpio nginx git gunicorn3 supervisor -y
```
Let me know if you see any errors with this or if the anything appears to actually get installed.  

Also, if you could send me the result of the 'pip list' command:

```bash
pip list
```

If Flask is not in the list you just printed out, you may have to run the `sudo pip install flask` command again.

**Reply by pseddonxvi** [reply]
I ran both commands, Flask is installed.
"0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded"
![image](https://github.com/nebhead/garage-zero/assets/146755532/87ac1e2e-fa5e-4961-8c05-683b6a69b229)

**Reply by nebhead** [reply]
OK, I haven't tested with Flask 3.x yet, so maybe we should bring this back to Flask 2.3.3.  To do this, you'll need to run:
```
sudo pip install flask==2.3.3
```
Also wanted to ask, when you go to just http://192.168.1.142 in your browser, what do you see?

**Reply by pseddonxvi** [reply]
Flask 2.3.3 installed successfully. When I visit http://192.168.1.142/ I get a 502 Bad Gateway error and some subtext saying "nginx/1.18.0"

**Reply by nebhead** [reply]
The 502 Gateway error might be a good thing here.  It means that nginx is up and configured properly.  It may just mean that the webapp script has crashed for some reason.  Maybe because it was using the new version of flask and something isn't compatible.  At this point we need to check logs.  

First, we need to check to see if supervisor is running at all.  I think you said the 9001 port wasn't working either, so maybe we need to check this from the command line.  

```bash
sudo supervisorctl
```

This should print out something like this: 

```
control                          RUNNING   pid 468, uptime 33 days, 1:29:44
webapp                           RUNNING   pid 469, uptime 33 days, 1:29:44
supervisor> 
```

Type `quit` + enter to exit.  

If you see something else, then maybe paste that in your response. 

OK, now onto the logs.  

Logs should be at `/home/pi/garage-zero/logs` and we should look for these files:

```
/home/pi/garage-zero/logs/webapp.err.log
/home/pi/garage-zero/logs/webapp.out.log
```

So use `cat` or `nano` to show us the contents:

```
cat /home/pi/garage-zero/logs/webapp.err.log
cat /home/pi/garage-zero/logs/webapp.out.log
```

They _might_ be big so look for relevant error information and copy/paste that into your response.  

Thanks!

**Reply by pseddonxvi** [reply]
The response for the first command was 
"unix:///var/run/supervisor.sock no such file"

Neither of the log files appeared to exist either. 
"cat: /home/pi/garage-zero/logs/webapp.err.log: No such file or directory"

**Comment by nebhead** [comment]
OK, looks like supervisor is not running.  

Also, please confirm you have a /home/pi/garage-zero folder, just for my sanity.  

Double check supervisor is installed:

```bash
sudo apt install supervisor
```

Then if installed:

```bash
sudo service supervisor start 
```

Then let me know what you see.

**Reply by pseddonxvi** [reply]
There is not a /home/pi/garage-zero folder, this has me pretty confused too.

Supervisor is installed, I ran the start command and it gave no errors.

**Comment by nebhead** [comment]
> There is not a /home/pi/garage-zero folder, this has me pretty confused too.

This is a hint.  Something happened with the git clone then, and this is why nothing is running properly.  

OK, for fun, let's try the following command and see if it works:

```bash
cd ~
git clone https://github.com/nebhead/garage-zero
```

BTW, is your username 'pi'?  For this version of garage-zero, it relies on the base username 'pi' and only works if using that directory.

**Reply by pseddonxvi** [reply]
Oh shoot! The username is not set to pi, I'd assume that's probably a big part of the issue.

**Reply by nebhead** [reply, marked as answer]
I think we cracked the case!  

So, the solution here would be to re-flash the OS and setup with the user 'pi' instead.  

BTW - There _is_ a newer version of the garage-zero software here: https://github.com/nebhead/garage-zero-v2 which does not use the pi user (if that's a problem for you) and installs to a globally accessible directory.

**Reply by pseddonxvi** [reply]
Thank you so much for the help!
